tuner/tuner/libtuner.py

```diff
@@ -221,6 +221,62 @@ def validate_devices(user_devices: list[str]) -> None:
         )
 
 
+def get_valid_benchmark_results(
+    benchmark_results: list[BenchmarkResult],
+) -> list[BenchmarkResult]:
+    """
+    Filter the benchmark_results list to return list with finite `time` values.
+    """
+    filtered_benchmark_results = [r for r in benchmark_results if math.isfinite(r.time)]
+    if len(filtered_benchmark_results) == 0:
+        logging.error("No successful candidate benchmarks.")
+
+    return filtered_benchmark_results
+
+
+def check_baseline_devices_uniqueness(baseline_results: list[BenchmarkResult]) -> bool:
```

Comment:
Based on the function name, we can't tell if `True` indicates a success or a failure. (That's why we have the `LogicalResult` type in LLVM, BTW!).

I think we could use this name instead:
```suggestion
def are_benchmark_devices_unique(baseline_results: list[BenchmarkResult]) -> bool:
```